agent/src/main/bin/testgrid-agent

```diff
@@ -0,0 +1,170 @@
+#!/bin/bash
+#
+# ----------------------------------------------------------------------------
+# Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+#
+# WSO2 Inc. licenses this file to you under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+# chkconfig: 345 99 05
+# description: TestGrid Agent daemon script
+#
+# A non-SUSE Linux start/stop script for Java daemons.
+#
+# Set this to Java installation
+JAVA_HOME=/usr/bin/
+
+scriptFile=$(readlink -fn $(type -p $0))             # the absolute, different path of this script file
+scriptDir=$(dirname $scriptFile)                     # absolute path of the script directory
+serviceNameLo="testgridAgent"                        # service name with the first letter in lowercase
+serviceName="TestGrid Agent"                         # service name
+serviceUser="root"                                   # OS user name for the service
+serviceGroup="root"                                  # OS group name for the service
+applDir="/opt/testgrid/agent"                        # home directory of the service application
+serviceUserHome="/home/$serviceUser"                 # home directory of the service user
+serviceLogFile="$applDir/$serviceNameLo.log"         # log file for StdOut/StdErr
+maxShutdownTime=15                                   # maximum number of seconds to wait for the daemon to terminate normally
+pidFile="$applDir/$serviceNameLo.pid"                # name of PID file (PID = process ID number)
+javaCommand="java"                                   # name of the Java launcher without the path
+javaArgs="-cp agent.jar:libs/* org.wso2.testgrid.agent.AgentApplication"  # arguments for Java launcher
```

Comment:
What were the assumptions to set the classpath as "agent.jar:libs/*"? For example, do you assume a testgrid dist should be there in this location?

Reply:
Going forward I think we should provider a single uber jar that we can just download and run. This allows easier setting up of slaves.